modules/custom/activity_creator/activity_creator.install

```diff
@@ -0,0 +1,117 @@
+<?php
+
+/**
+ * @file
+ * Installation code for the activity_creator module.
+ */
+
+use Drupal\Core\Database\Database;
+
+/**
+ * Implements hook_schema().
+ */
+function activity_creator_schema() {
+  $schema['activity_notification_status'] = [
+    'fields' => [
+      'uid' => [
+        'description' => 'The {user}.uid of user.',
+        'type' => 'int',
+        'unsigned' => TRUE,
+        'not null' => TRUE,
+      ],
+      'aid' => [
+        'description' => 'The {activity}.id of activity.',
+        'type' => 'int',
+        'unsigned' => TRUE,
+        'not null' => TRUE,
+      ],
+      'status' => [
+        'description' => 'The activity status.',
+        'type' => 'int',
+        'unsigned' => TRUE,
+        'not null' => TRUE,
+        'default' => 1,
+      ],
+    ],
+    'indexes' => [
+      'ans_uid' => ['uid'],
+      'ans_aid' => ['aid'],
+      'ans_uid_aid' => ['uid', 'aid'],
+      'ans_status' => ['status'],
+    ],
+  ];
+
+  return $schema;
+}
+
+/**
+ * Create new database table {activity_notification_status}.
+ */
+function activity_creator_update_8001() {
+  $spec['activity_notification_status'] = [
+    'fields' => [
+      'uid' => [
+        'description' => 'The {user}.uid of user.',
+        'type' => 'int',
+        'unsigned' => TRUE,
+        'not null' => TRUE,
+      ],
+      'aid' => [
+        'description' => 'The {activity}.id of activity.',
+        'type' => 'int',
+        'unsigned' => TRUE,
+        'not null' => TRUE,
+      ],
+      'status' => [
+        'description' => 'The activity status.',
+        'type' => 'int',
+        'unsigned' => TRUE,
+        'not null' => TRUE,
+        'default' => 1,
+      ],
+    ],
+    'indexes' => [
+      'ans_uid' => ['uid'],
+      'ans_aid' => ['aid'],
+      'ans_uid_aid' => ['uid', 'aid'],
+      'ans_status' => ['status'],
+    ],
+  ];
+
+  $db = Database::getConnection();
+  $schema = $db->schema();
+  $schema->createTable('activity_notification_status', $spec['activity_notification_status']);
+}
+
+/**
+ * Migrate all the activity information to new table.
+ * This is necessary as we have changed the logic of reading notifications
+ * and marking them as seen. So, we have migrate the existing activity entries
+ * to new table so as to avoid any missing notifications by users.
+ *
+ * @throws \Exception
+ */
+function activity_creator_update_8002() {
```

Comment:
@navneet0693 this is a nice one to do in a post update hook.

Rule of thumb is that we run data migrations in post update hooks instead of normal update hooks.

Reply:
Ok, I will convert it to batch and move to post_update_hook.